src/PaymentGateways/Gateways/Stripe/StripePaymentElementGateway/StripePaymentElementGatewaySubscriptionModule.php

```diff
@@ -105,6 +105,10 @@ public function createSubscription(
         return new RespondToBrowser([
             'clientSecret' => $stripeSubscription->latest_invoice->payment_intent->client_secret,
             'returnUrl' => $stripeGatewayData->successUrl,
+            'billingDetails' => [
+                'name' => trim("$donation->firstName $donation->lastName"),
+                'email' => $donation->email
+            ],
```

Comment:
Does this change have any relationship with the currency switcher?

Reply:
Nope! moved this to develop 3c3f97e5